samza-core/src/main/java/org/apache/samza/container/grouper/task/TaskAssignmentManager.java

```diff
@@ -131,12 +141,12 @@ public void writeTaskContainerMapping(String taskName, String containerId) {
    */
   public void deleteTaskContainerMappings(Iterable<String> taskNames) {
     for (String taskName : taskNames) {
-      metadataStore.delete(taskName);
+      taskContainerMappingMetadataStore.delete(taskName);
       taskNameToContainerId.remove(taskName);
     }
   }
 
   public void close() {
-    metadataStore.close();
+    taskContainerMappingMetadataStore.close();
```

Comment:
I think it will be better to close the other store as well.

Reply:
done